README
NAME
    Net::SNMP::Mixin::IpCidrRouteTable - mixin class for the mib-II
    ipCidrRouteTable

VERSION
    Version 0.04

SYNOPSIS
      use Net::SNMP;
      use Net::SNMP::Mixin;

      #...

      my $session = Net::SNMP->session( -hostname => 'foo.bar.com' );

      $session->mixer('Net::SNMP::Mixin::IpCidrRouteTable');
      $session->init_mixins;
      snmp_dispatcher();

      die $session->errors if $session->errors;

      my @routes = $session->get_ip_cidr_route_table();

      foreach my $route (@routes) {
          my $dest     = $route->{ipCidrRouteDest};
          my $mask     = $route->{ipCidrRouteMask};
          my $tos      = $route->{ipCidrRouteTos};
          my $next_hop = $route->{ipCidrRouteNextHop};
          my $if_index = $route->{ipCidrRouteIfIndex};

          print "$dest/$mask/$tos/$next_hop => $if_index/$next_hop\n";
      }

DESCRIPTION
    A Net::SNMP mixin class for mib-II ipCidrRouteTable info.

MIXIN METHODS
  OBJ->get_ip_cidr_route_table()
    Returns a sorted list of mib-II ipCidrRouteTable entries. Every list
    element (route entry) is a hashref with the following fields and values:

        {
            ipCidrRouteDest         => IpAddress,           # tbl index
            ipCidrRouteMask         => IpAddress,           # tbl index
            ipCidrRouteTos          => Integer32,           # tbl index
            ipCidrRouteNextHop      => IpAddress,           # tbl index
            ipCidrRouteIfIndex      => Integer32,
            ipCidrRouteType         => INTEGER,
            ipCidrRouteTypeString   => String,              # resolved enum
            ipCidrRouteProto        => INTEGER,
            ipCidrRouteProtoString  => String,              # resolved enum
            ipCidrRouteAge          => Integer32,
            ipCidrRouteInfo         => OBJECT IDENTIFIER,
            ipCidrRouteNextHopAS    => Integer32,
            ipCidrRouteMetric1      => Integer32,
            ipCidrRouteMetric2      => Integer32,
            ipCidrRouteMetric3      => Integer32,
            ipCidrRouteMetric4      => Integer32,
            ipCidrRouteMetric5      => Integer32,
            ipCidrRouteStatus       => RowStatus,
            ipCidrRouteStatusString => String,              # resolved enum
        }

INITIALIZATION
  OBJ->_init($reload)
    Fetch the mib-II ipCidrRouteTable from the host. Don't call this method
    direct!

PRIVATE METHODS
    Only for developers or maintainers.

  _fetch_ip_cidr_route_tbl($session)
    Fetch the ipCidrRouteTable once during object initialization.

  _ip_cidr_route_tbl_cb($session)
    The callback for _fetch_ip_cidr_route_tbl().

SEE ALSO
    Net::SNMP::Mixin

REQUIREMENTS
    Net::SNMP, Net::SNMP::Mixin

BUGS, PATCHES & FIXES
    There are no known bugs at the time of this release. However, if you
    spot a bug or are experiencing difficulties that are not explained
    within the POD documentation, please submit a bug to the RT system (see
    link below). However, it would help greatly if you are able to pinpoint
    problems or even supply a patch.

    Fixes are dependant upon their severity and my availablity. Should a fix
    not be forthcoming, please feel free to (politely) remind me by sending
    an email to gaissmai@cpan.org .

      RT: http://rt.cpan.org/Public/Dist/Display.html?Name=Net-SNMP-Mixin-IpCidrRouteTable

AUTHOR
    Karl Gaissmaier <karl.gaissmaier at uni-ulm.de>

COPYRIGHT & LICENSE
    Copyright 2011 Karl Gaissmaier, all rights reserved.

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.



Changes
Revision history for Net-SNMP-Mixin-IpCidrRouteTable

0.04	Tue, 15 Oct 2013 16:21:50 +0200
	* fix POD typo [rt.cpan.org #89522]
	* tweak Build.pl

0.03	Sat, 12 Nov 2011 23:10:59 +0100
	* fix LICENSE file

0.02	Sat, 12 Nov 2011 22:32:45 +0100
	* return table in oid lexical sort order

0.01	Sat, 12 Nov 2011 18:52:25 +0100
	* first CPAN version


Build.PL
use strict;
use warnings;
use Module::Build;

my $builder = Module::Build->new(
  module_name       => 'Net::SNMP::Mixin::IpCidrRouteTable',
  create_readme     => 1,
  license           => 'perl',
  dist_author       => 'Karl Gaissmaier <karl.gaissmaier (at) uni-ulm.de>',
  dist_version_from => 'lib/Net/SNMP/Mixin/IpCidrRouteTable.pm',
  requires          => {
    'perl'             => '5.6.0',
    'Net::SNMP'        => '0',
    'Net::SNMP::Mixin' => '0.11',
    'Sub::Exporter'    => '0.978',
  },
  build_requires => {
    'Module::Build' => '0',
    'Test::More'    => '0',
  },
  create_makefile_pl => 'small',
  add_to_cleanup => [qw/Net-SNMP-Mixin-* Makefile* META.yml README/],
);

# -----------------------------------------------------------------------
# shall we perform live tests?

print "==> For running live test you need a network connection <==\n";
if (
  $builder->y_n(
    'Do you want to run live tests against an active SNMP agent?', 'n'
  )
  )
{
  my $snmp_agent =
    $builder->prompt('Enter a hostname/ip-address of snmp agent []:');
  $builder->notes( snmp_agent => $snmp_agent || '' );

  my $community = $builder->prompt('Enter read-only community [public]:');
  $builder->notes( snmp_community => $community || 'public' );

  my $version = $builder->prompt('Enter snmp version [2c]:');
  $builder->notes( snmp_version => $version || '2c' );
}
else {
  print "Skipping live tests.\n";
  $builder->notes( snmp_agent     => '' );
  $builder->notes( snmp_community => '' );
  $builder->notes( snmp_version   => '' );
}

# -----------------------------------------------------------------------

$builder->create_build_script();


examples/ip_cidr_rtbl.pl
#!/usr/bin/env perl

use strict;
use warnings;

=head1 NAME

iprtbl.pl

=head1 ABSTRACT

A script to get the MIB-2 ipCidrRouteTable information from devices supporting the MIBs.

=head1 SYNOPSIS

 ip_cidr_rtbl.pl OPTIONS agent agent ...

 ip_cidr_rtbl.pl OPTIONS -i <agents.txt

=head2 OPTIONS

  -c snmp_community
  -v snmp_version
  -t snmp_timeout
  -r snmp_retries

  -d			Net::SNMP debug on
  -i			read agents from stdin, one agent per line
  -B			nonblocking

=cut

use blib;
use Net::SNMP qw(:debug :snmp);
use Net::SNMP::Mixin;

use Getopt::Std;

my %opts;
getopts( 'iBdt:r:c:v:', \%opts ) or usage();

my $debug       = $opts{d} || undef;
my $community   = $opts{c} || 'public';
my $version     = $opts{v} || '2';
my $nonblocking = $opts{B} || 0;
my $timeout     = $opts{t} || 5;
my $retries     = $opts{t} || 0;

my $from_stdin = $opts{i} || undef;

my @agents = @ARGV;
push @agents, <STDIN> if $from_stdin;
chomp @agents;
usage('missing agents') unless @agents;

my @sessions;
foreach my $agent ( sort @agents ) {
    my ( $session, $error ) = Net::SNMP->session(
        -community   => $community,
        -hostname    => $agent,
        -version     => $version,
        -nonblocking => $nonblocking,
        -timeout     => $timeout,
        -retries     => $retries,
        -debug       => $debug ? DEBUG_ALL : 0,
    );

    if ($error) {
        warn $error;
        next;
    }

    $session->mixer(qw/ Net::SNMP::Mixin::IpCidrRouteTable /);

    $session->init_mixins;
    push @sessions, $session;

}
snmp_dispatcher();

# warn on errors during initialization
foreach my $session (@sessions) {
    if ( $session->errors ) {
        foreach my $error ( $session->errors ) {
            warn $session->hostname . ": $error\n";
        }
    }
}

# remove sessions with errors from the sessions list
@sessions = grep { not $_->errors(1) } @sessions;

foreach my $session ( sort { $a->hostname cmp $b->hostname } @sessions ) {
    print_ip_cidr_rtbl($session);
}

exit 0;

###################### end of main ######################

sub print_ip_cidr_rtbl {
    my $session = shift;

    print "\n";
    printf "Hostname: %-15.15s\n", $session->hostname;

    print '-' x 66, "\n";
    printf "%-18s => %-15s %8s %10s %8s\n", 'Route', 'NextHop',
      'Proto', 'Type', 'Metric1';
    print '-' x 66, "\n";

    my @routes = $session->get_ip_cidr_route_table();

    foreach my $route (@routes) {
        my $dest      = $route->{ipCidrRouteDest};
        my $mask      = $route->{ipCidrRouteMask};
        my $bits      = mk_bits($mask);
        my $nhop      = $route->{ipCidrRouteNextHop};
        my $proto_str = $route->{ipCidrRouteProtoString};
        my $type_str  = $route->{ipCidrRouteTypeString};
        my $metric1   = $route->{ipCidrRouteMetric1};

        printf "%-18s => %-15s %8s %10s %8s\n", "$dest/$bits", $nhop,
          $proto_str, $type_str, $metric1;
    }
}

# convert IPv4 netmasks into cidr notation, eg: '255.255.255.0' => 24
sub mk_bits {

    # sorry, highly condensed code, write once read never
    return unpack( '%B*', pack( 'C4', split( /\./, shift ) ) );
}

sub usage {
    my @msg = @_;
    die <<EOT;
>>>>>> @msg
    Usage: $0 [options] hostname
   
    	-c community
  	-v version
  	-t timeout
  	-r retries
  	-d		Net::SNMP debug on
	-i		read agents from stdin
  	-B		nonblocking
EOT
}

=head1 AUTHOR

Karl Gaissmaier, karl.gaissmaier (at) uni-ulm.de

=head1 COPYRIGHT

Copyright (C) 2011 by Karl Gaissmaier

This program is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.

=cut

# vim: sw=2


lib/Net/SNMP/Mixin/IpCidrRouteTable.pm
package Net::SNMP::Mixin::IpCidrRouteTable;

use strict;
use warnings;

#
# store this package name in a handy variable,
# used for unambiguous prefix of mixin attributes
# storage in object hash
#
my $prefix = __PACKAGE__;

#
# this module import config
#
use Carp ();
use Net::SNMP qw(oid_lex_sort);
use Net::SNMP::Mixin::Util qw/idx2val push_error/;

#
# this module export config
#
my @mixin_methods;

BEGIN {
  @mixin_methods = (qw/ get_ip_cidr_route_table /);
}

use Sub::Exporter -setup => {
  exports => [@mixin_methods],
  groups  => { default => [@mixin_methods], },
};

#
# SNMP oid constants used in this module
#
# from IP Forwarding Table MIB, RFC 2096
#
use constant {
  IP_CIDR_ROUTE_TABLE => '1.3.6.1.2.1.4.24.4',
  IP_CIDR_ROUTE_ENTRY => '1.3.6.1.2.1.4.24.4.1',

  IP_CIDR_ROUTE_DEST      => '1.3.6.1.2.1.4.24.4.1.1',
  IP_CIDR_ROUTE_MASK      => '1.3.6.1.2.1.4.24.4.1.2',
  IP_CIDR_ROUTE_TOS       => '1.3.6.1.2.1.4.24.4.1.3',
  IP_CIDR_ROUTE_NEXTHOP   => '1.3.6.1.2.1.4.24.4.1.4',
  IP_CIDR_ROUTE_IFINDEX   => '1.3.6.1.2.1.4.24.4.1.5',
  IP_CIDR_ROUTE_TYPE      => '1.3.6.1.2.1.4.24.4.1.6',
  IP_CIDR_ROUTE_PROTO     => '1.3.6.1.2.1.4.24.4.1.7',
  IP_CIDR_ROUTE_AGE       => '1.3.6.1.2.1.4.24.4.1.8',
  IP_CIDR_ROUTE_INFO      => '1.3.6.1.2.1.4.24.4.1.9',
  IP_CIDR_ROUTE_NEXTHOPAS => '1.3.6.1.2.1.4.24.4.1.10',
  IP_CIDR_ROUTE_METRIC1   => '1.3.6.1.2.1.4.24.4.1.11',
  IP_CIDR_ROUTE_METRIC2   => '1.3.6.1.2.1.4.24.4.1.12',
  IP_CIDR_ROUTE_METRIC3   => '1.3.6.1.2.1.4.24.4.1.13',
  IP_CIDR_ROUTE_METRIC4   => '1.3.6.1.2.1.4.24.4.1.14',
  IP_CIDR_ROUTE_METRIC5   => '1.3.6.1.2.1.4.24.4.1.15',
  IP_CIDR_ROUTE_STATUS    => '1.3.6.1.2.1.4.24.4.1.16',
};

#
# the ipCidrRouteType enum
#
my %type_enum = (
  1 => 'other',
  2 => 'reject',
  3 => 'local',
  4 => 'remote',
);

#
# the ipCidrRouteProto enum
#
my %proto_enum = (
  1  => 'other',
  2  => 'local',
  3  => 'netmgmt',
  4  => 'icmp',
  5  => 'egp',
  6  => 'ggp',
  7  => 'hello',
  8  => 'rip',
  9  => 'is-is',
  10 => 'es-is',
  11 => 'ciscoIgrp',
  12 => 'bbnSpfIgp',
  13 => 'ospf',
  14 => 'bgp',
  15 => 'idpr',
  16 => 'ciscoEigrp',
);

#
# the ipCidrRouteStatus enum
#
my %status_enum = (
  1 => 'active',
  2 => 'notInService',
  3 => 'notReady',
  4 => 'createAndGo',
  5 => 'createAndWait',
  6 => 'destroy',
);

=head1 NAME

Net::SNMP::Mixin::IpCidrRouteTable - mixin class for the mib-II ipCidrRouteTable

=head1 VERSION

Version 0.04

=cut

our $VERSION = '0.04';

=head1 SYNOPSIS

  use Net::SNMP;
  use Net::SNMP::Mixin;

  #...

  my $session = Net::SNMP->session( -hostname => 'foo.bar.com' );

  $session->mixer('Net::SNMP::Mixin::IpCidrRouteTable');
  $session->init_mixins;
  snmp_dispatcher();

  die $session->errors if $session->errors;

  my @routes = $session->get_ip_cidr_route_table();

  foreach my $route (@routes) {
      my $dest     = $route->{ipCidrRouteDest};
      my $mask     = $route->{ipCidrRouteMask};
      my $tos      = $route->{ipCidrRouteTos};
      my $next_hop = $route->{ipCidrRouteNextHop};
      my $if_index = $route->{ipCidrRouteIfIndex};

      print "$dest/$mask/$tos/$next_hop => $if_index/$next_hop\n";
  }

=head1 DESCRIPTION

A Net::SNMP mixin class for mib-II ipCidrRouteTable info.

=head1 MIXIN METHODS

=head2 B<< OBJ->get_ip_cidr_route_table() >>

Returns a sorted list of mib-II ipCidrRouteTable entries. Every list element (route entry) is a hashref with the following fields and values:

    {
        ipCidrRouteDest         => IpAddress,           # tbl index
        ipCidrRouteMask         => IpAddress,           # tbl index
        ipCidrRouteTos          => Integer32,           # tbl index
        ipCidrRouteNextHop      => IpAddress,           # tbl index
        ipCidrRouteIfIndex      => Integer32,
        ipCidrRouteType         => INTEGER,
        ipCidrRouteTypeString   => String,              # resolved enum
        ipCidrRouteProto        => INTEGER,
        ipCidrRouteProtoString  => String,              # resolved enum
        ipCidrRouteAge          => Integer32,
        ipCidrRouteInfo         => OBJECT IDENTIFIER,
        ipCidrRouteNextHopAS    => Integer32,
        ipCidrRouteMetric1      => Integer32,
        ipCidrRouteMetric2      => Integer32,
        ipCidrRouteMetric3      => Integer32,
        ipCidrRouteMetric4      => Integer32,
        ipCidrRouteMetric5      => Integer32,
        ipCidrRouteStatus       => RowStatus,
        ipCidrRouteStatusString => String,              # resolved enum
    }

=cut

sub get_ip_cidr_route_table {
  my $session = shift;
  Carp::croak "'$prefix' not initialized,"
    unless $session->{$prefix}{__initialized};

  # stash for return values
  my @route_tbl;

  my (
    $ipCidrRouteDest,        $ipCidrRouteMask,
    $ipCidrRouteTos,         $ipCidrRouteNextHop,
    $ipCidrRouteIfIndex,     $ipCidrRouteType,
    $ipCidrRouteTypeString,  $ipCidrRouteProto,
    $ipCidrRouteProtoString, $ipCidrRouteAge,
    $ipCidrRouteInfo,        $ipCidrRouteNextHopAS,
    $ipCidrRouteMetric1,     $ipCidrRouteMetric2,
    $ipCidrRouteMetric3,     $ipCidrRouteMetric4,
    $ipCidrRouteMetric5,     $ipCidrRouteStatus,
    $ipCidrRouteStatusString,
  );

  #
  # index is ipCidrRouteDest,ipCidrRouteMask,ipCidrRouteTos,ipCidrRouteNextHop
  #
  foreach
    my $idx ( oid_lex_sort keys %{ $session->{$prefix}{ipCidrRouteDest} } )
  {
    $ipCidrRouteDest    = $session->{$prefix}{ipCidrRouteDest}{$idx};
    $ipCidrRouteMask    = $session->{$prefix}{ipCidrRouteMask}{$idx};
    $ipCidrRouteTos     = $session->{$prefix}{ipCidrRouteTos}{$idx};
    $ipCidrRouteNextHop = $session->{$prefix}{ipCidrRouteNextHop}{$idx};
    $ipCidrRouteIfIndex = $session->{$prefix}{ipCidrRouteIfIndex}{$idx};

    $ipCidrRouteType = $session->{$prefix}{ipCidrRouteType}{$idx} || -1;
    $ipCidrRouteTypeString = $type_enum{$ipCidrRouteType} || 'unknown';

    $ipCidrRouteProto = $session->{$prefix}{ipCidrRouteProto}{$idx} || -1;
    $ipCidrRouteProtoString = $proto_enum{$ipCidrRouteProto} || 'unknown';

    $ipCidrRouteAge       = $session->{$prefix}{ipCidrRouteAge}{$idx};
    $ipCidrRouteInfo      = $session->{$prefix}{ipCidrRouteInfo}{$idx};
    $ipCidrRouteNextHopAS = $session->{$prefix}{ipCidrRouteNextHopAS}{$idx};
    $ipCidrRouteMetric1   = $session->{$prefix}{ipCidrRouteMetric1}{$idx};
    $ipCidrRouteMetric2   = $session->{$prefix}{ipCidrRouteMetric2}{$idx};
    $ipCidrRouteMetric3   = $session->{$prefix}{ipCidrRouteMetric3}{$idx};
    $ipCidrRouteMetric4   = $session->{$prefix}{ipCidrRouteMetric4}{$idx};
    $ipCidrRouteMetric5   = $session->{$prefix}{ipCidrRouteMetric5}{$idx};

    $ipCidrRouteStatus = $session->{$prefix}{ipCidrRouteStatus}{$idx} || -1;
    $ipCidrRouteStatusString = $status_enum{$ipCidrRouteStatus} || 'unknown';

    push @route_tbl,
      {
      ipCidrRouteDest         => $ipCidrRouteDest,
      ipCidrRouteMask         => $ipCidrRouteMask,
      ipCidrRouteTos          => $ipCidrRouteTos,
      ipCidrRouteNextHop      => $ipCidrRouteNextHop,
      ipCidrRouteIfIndex      => $ipCidrRouteIfIndex,
      ipCidrRouteType         => $ipCidrRouteType,
      ipCidrRouteTypeString   => $ipCidrRouteTypeString,
      ipCidrRouteProto        => $ipCidrRouteProto,
      ipCidrRouteProtoString  => $ipCidrRouteProtoString,
      ipCidrRouteAge          => $ipCidrRouteAge,
      ipCidrRouteInfo         => $ipCidrRouteInfo,
      ipCidrRouteNextHopAS    => $ipCidrRouteNextHopAS,
      ipCidrRouteMetric1      => $ipCidrRouteMetric1,
      ipCidrRouteMetric2      => $ipCidrRouteMetric2,
      ipCidrRouteMetric3      => $ipCidrRouteMetric3,
      ipCidrRouteMetric4      => $ipCidrRouteMetric4,
      ipCidrRouteMetric5      => $ipCidrRouteMetric5,
      ipCidrRouteStatus       => $ipCidrRouteStatus,
      ipCidrRouteStatusString => $ipCidrRouteStatusString,
      };
  }

  return @route_tbl;
}

=head1 INITIALIZATION

=head2 B<< OBJ->_init($reload) >>

Fetch the mib-II ipCidrRouteTable from the host. Don't call this method direct!

=cut

sub _init {
  my ( $session, $reload ) = @_;

  die "$prefix already initalized and reload not forced.\n"
    if $session->{$prefix}{__initialized} && not $reload;

  # populate the object with needed mib values
  #
  # initialize the object for ipCidrRouteTable infos
  _fetch_ip_cidr_route_tbl($session);
  return if $session->error;

  return 1;
}

=head1 PRIVATE METHODS

Only for developers or maintainers.

=head2 B<< _fetch_ip_cidr_route_tbl($session) >>

Fetch the ipCidrRouteTable once during object initialization.

=cut

sub _fetch_ip_cidr_route_tbl {
  my $session = shift;
  my $result;

  # fetch the ipCidrRouteTable
  $result = $session->get_table(
    -baseoid => IP_CIDR_ROUTE_TABLE,

    # define callback if in nonblocking mode
    $session->nonblocking ? ( -callback => \&_ip_cidr_route_tbl_cb ) : (),
  );

  unless ( defined $result ) {

    # Net::SNMP looses sometimes error messages in nonblocking
    # mode, so we save them in an extra buffer
    my $err_msg = $session->error;
    push_error( $session, "$prefix: $err_msg" ) if $err_msg;
    return;
  }

  # in nonblocking mode the callback will be called asynchronously
  return 1 if $session->nonblocking;

  # ok we are in synchronous mode, call the result mangling function
  # by hand
  _ip_cidr_route_tbl_cb($session);

}

=head2 B<< _ip_cidr_route_tbl_cb($session) >>

The callback for _fetch_ip_cidr_route_tbl().

=cut

sub _ip_cidr_route_tbl_cb {
  my $session = shift;
  my $vbl     = $session->var_bind_list;

  unless ( defined $vbl ) {

    # Net::SNMP looses sometimes error messages in nonblocking
    # mode, so we save them in an extra buffer
    my $err_msg = $session->error;
    push_error( $session, "$prefix: $err_msg" ) if $err_msg;
    return;
  }

  # build parallel hashes, keys are the index:
  #
  # ipCidrRouteDest,ipCidrRouteMask,ipCidrRouteTos,ipCidrRouteNextHop
  #

  $session->{$prefix}{ipCidrRouteDest} =
    idx2val( $vbl, IP_CIDR_ROUTE_DEST, undef, undef, );

  $session->{$prefix}{ipCidrRouteMask} =
    idx2val( $vbl, IP_CIDR_ROUTE_MASK, undef, undef, );

  $session->{$prefix}{ipCidrRouteTos} =
    idx2val( $vbl, IP_CIDR_ROUTE_TOS, undef, undef, );

  $session->{$prefix}{ipCidrRouteNextHop} =
    idx2val( $vbl, IP_CIDR_ROUTE_NEXTHOP, undef, undef, );

  $session->{$prefix}{ipCidrRouteIfIndex} =
    idx2val( $vbl, IP_CIDR_ROUTE_IFINDEX, undef, undef, );

  $session->{$prefix}{ipCidrRouteType} =
    idx2val( $vbl, IP_CIDR_ROUTE_TYPE, undef, undef, );

  $session->{$prefix}{ipCidrRouteProto} =
    idx2val( $vbl, IP_CIDR_ROUTE_PROTO, undef, undef, );

  $session->{$prefix}{ipCidrRouteAge} =
    idx2val( $vbl, IP_CIDR_ROUTE_AGE, undef, undef, );

  $session->{$prefix}{ipCidrRouteInfo} =
    idx2val( $vbl, IP_CIDR_ROUTE_INFO, undef, undef, );

  $session->{$prefix}{ipCidrRouteNextHopAS} =
    idx2val( $vbl, IP_CIDR_ROUTE_NEXTHOPAS, undef, undef, );

  $session->{$prefix}{ipCidrRouteMetric1} =
    idx2val( $vbl, IP_CIDR_ROUTE_METRIC1, undef, undef, );

  $session->{$prefix}{ipCidrRouteMetric2} =
    idx2val( $vbl, IP_CIDR_ROUTE_METRIC2, undef, undef, );

  $session->{$prefix}{ipCidrRouteMetric3} =
    idx2val( $vbl, IP_CIDR_ROUTE_METRIC3, undef, undef, );

  $session->{$prefix}{ipCidrRouteMetric4} =
    idx2val( $vbl, IP_CIDR_ROUTE_METRIC4, undef, undef, );

  $session->{$prefix}{ipCidrRouteMetric5} =
    idx2val( $vbl, IP_CIDR_ROUTE_METRIC5, undef, undef, );

  $session->{$prefix}{ipCidrRouteStatus} =
    idx2val( $vbl, IP_CIDR_ROUTE_STATUS, undef, undef, );

  $session->{$prefix}{__initialized}++;
}

unless ( caller() ) {
  print "$prefix compiles and initializes successful.\n";
}

=head1 SEE ALSO

L<< Net::SNMP::Mixin >>

=head1 REQUIREMENTS

L<< Net::SNMP >>, L<< Net::SNMP::Mixin >>

=head1 BUGS, PATCHES & FIXES

There are no known bugs at the time of this release. However, if you spot a bug or are experiencing difficulties that are not explained within the POD documentation, please submit a bug to the RT system (see link below). However, it would help greatly if you are able to pinpoint problems or even supply a patch. 

Fixes are dependant upon their severity and my availablity. Should a fix not be forthcoming, please feel free to (politely) remind me by sending an email to gaissmai@cpan.org .

  RT: http://rt.cpan.org/Public/Dist/Display.html?Name=Net-SNMP-Mixin-IpCidrRouteTable


=head1 AUTHOR

Karl Gaissmaier <karl.gaissmaier at uni-ulm.de>

=head1 COPYRIGHT & LICENSE

Copyright 2011 Karl Gaissmaier, all rights reserved.

This program is free software; you can redistribute it and/or modify it
under the same terms as Perl itself.

=cut

1;

# vim: sw=2
